@xen-orchestra/web-core/lib/components/PowerStateIcon.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <UiIcon :class="className" :icon="icon" class="power-state-icon" />
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import type { POWER_STATE } from '@core/types/power-state.type'
+import { faMoon, faPause, faPlay, faQuestion, faStop } from '@fortawesome/free-solid-svg-icons'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  state: POWER_STATE
+}>()
+
+const icons = {
+  running: faPlay,
+  paused: faPause,
+  suspended: faMoon,
+  halted: faStop,
+}
+
+const icon = computed(() => icons[props.state] ?? faQuestion)
+
+const className = computed(() => `state-${props.state.toLocaleLowerCase()}`)
```

Comment:
Since `state` is typed, it's not necessary:
1. to have a default icon
2. to change the case

```suggestion
const icon = computed(() => icons[props.state] ?? faQuestion)

const className = computed(() => `state-${props.state}`)
```